### ISSUE TITLE
what's the default option for retweets? How to identify original/retweet?

### ISSUE THREAD

**Issue by Cheryl520** [twintproject/twint#791]
Since I want to also include the retweets in my search, but i forgot to set up 'retweets=True' in the configuration, I'm wondering if i didn't set anything for retweet-related options in the configuration, the tweets that i get include or exclude the retweets?
I found all the 'retweet' column is False in my file. Does that mean the default is that only original tweets are scraped?
Or it's because my search range is 2018-01-01 to 2020-01-01 ( search tweets containing certain keyword), and the retweets are too old so it cannot be returned? (I saw issues posted before saying that only recent retweets can be returned).

I tried 'Native_retweets'=True to scrape only retweets again in that range, however, no tweets are returned.

I also saw  #630, and noticed that there's new style and old style retweets. So i want to know if the tweets that i scraped without setting any retweets options returns actually with old style retweets (which can be filtered by searching for 'RT @'  ), but exclude the new style retweets (which is impossible to get because they are too old)? 

In summary, I want to know how to identify retweets/original in my scrapped file.

Thank you in advance!

**Comment by konnycode** [comment]
Native retweets and commented retweets are a whole different story.

The "native retweets"-option can only collect native retweets for the last week or so. The only other option to collect native retweets (as far as i know!) is slow profile-scraping with "profile_full" set on True. However, the "scraping the profile"-function was broken last time I checked.

The search-method only collects commented retweets. They are stored in the "quote_url" column. So with pandas in python you could do this:

`df.loc[df['quote_url'].notna(), 'retweet'] = True`

This will set the 'retweet'-value on True everywhere you have a quoted twitter status. For further analysis like cross reference you can parse the twitter urls and extract the quoted twitter-usernames and statuses.

`df.loc[:,'rt_account'] = df['quote_url'].apply(lambda x: x.split('/')[3])
df.loc[:,'rt_status'] = df['quote_url'].apply(lambda x: x.split('/')[5]).astype('int64')`

Hope this helps

**Comment by Cheryl520** [comment]
> Native retweets and commented retweets are a whole different story.
> The "native retweets"-option can only collect native retweets for the last week or so. The only other option to collect native retweets (as far as i know!) is slow profile-scraping with "profile_full" set on True. However, the "scraping the profile"-method was broken last time I checked.
> The search-method only collects commented retweets. They are stored in the "quote_url" column. So with pandas in python you could do this:
> df.loc[df['quote_url'].notna(), 'retweet'] = True
> This will set the 'retweet'-value on True everywhere you have a quoted twitter status. For further analysis like cross reference you can parse the twitter urls and extract the quoted twitter-usernames and statuses.
> df.loc[:,'rt_account'] = df['quote_url'].apply(lambda x: x.split('/')[3]) df.loc[:,'rt_status'] = df['quote_url'].apply(lambda x: x.split('/')[5]).astype('int64')
> Hope this helps

Thanks! it helps a lot
